File: ch09/App_Code/CS/CustomerRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

/// <summary>
/// Summary description for CustomerRepository
/// </summary>
public  class CustomerRepository
{
    CustomerContext _context = new CustomerContext();

    public IEnumerable<Customer> SelectCustomers([System.Web.ModelBinding.QueryString] int? id)
    {
        if (id.HasValue)
            return _context.Customer.Where(c => c.ID == id).AsEnumerable();
        else
            return _context.Customer.AsEnumerable();
    }
}

File: ch09/App_Code/VB/CustomerRepository.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace VB
    ''' <summary>
    ''' Summary description for CustomerRepository
    ''' </summary>
    Public Class CustomerRepository
        Private _context As New CustomerContext()

        Public Function SelectCustomers(<System.Web.ModelBinding.QueryString> id As System.Nullable(Of Integer)) As IEnumerable(Of Customer)
            If id.HasValue Then
                Return _context.Customer.Where(Function(c) c.ID = id).AsEnumerable()
            Else
                Return _context.Customer.AsEnumerable()
            End If
        End Function
    End Class
End Namespace